sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_connection.py

```diff
@@ -90,6 +90,9 @@ def __init__(self, endpoint, **kwargs):
         # type(str, Any) -> None
         parsed_url = urlparse(endpoint)
         self._hostname = parsed_url.hostname
+        custom_parsed_url = urlparse(kwargs.get("custom_endpoint_address"))
+        self.custom_hostname = custom_parsed_url.hostname
+        self.custom_port = custom_parsed_url.port
```

Comment:
This assumes that a custom endpoint will always be supplied?

Reply:
Also - why are these being assigned to `self` when they just get passed straight into the transport?